source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,127 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+            }
+        }
+
+For the MedcoMD system, we will create JSON objects for each field and paste them into the properties 
+object of the JSON Schema above.
+
+
+.. code::
+
+    // Deterministic encryption supports querying and is equally as secure as random encryption for high cardinality (very unique) fields such as SSN. A privileged user in the MedcoMD system might want to query on SSN to verify a patient's identity. One thing to note is that you do not have to include keyId if you specified it in the top-level encryptMetadata field. Though you could include it if you want different users with specific keys to be able to access a specific subset of the encrypted data.
+
+    "ssn": {
+        "encrypt": {
+            "bsonType": "int",
+            "keyId": <keyID>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+        }
+    }
+
+
+.. code:: javascript
+
+    // Random encryption does not support filtering and is best used on fields with low cardinality (very few unique fields) to prevent against frequency analysis attacks. Since there are only 8 blood types, this field will be randomly encrypted. To decrypt this data, a privileged user would have to query on another field via a secure collection, and this data would be returned unencrypted.
+
+    "bloodType": {
+        "encrypt": {
+            "bsonType": "string",
+            "keyId": <keyId>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+        }
+    }
+
+
+.. code:: javascript
+
+    // Arrays can only be auto-encrypted via random encryption. 
+
+    "medicalRecords": {
+        "encrypt": {
+            "bsonType": "array",
+            "keyId": <keyID>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+        }
+    }
+
+.. code:: javascript
+
+    // insurance.policyNumber number is deterministically encrypted because the policy number is a very unique field. Note, that the JSON for policyNumber follows the same structure the JSON for SSN, however, policyNumber is nested inside insurance.properties because it is a nested field.
+
+    "insurance": {
+        "bsonType": "object",
+        "properties": {
+            "policyNumber": {
+                "encrypt": {
+                    "bsonType": "int",
+                    "keyId": <keyID>,
+                    "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                }
+            }
+        }
+    }    
+
+
+
+- View the full JSON Schema for this MedcoMD guide `here <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-json-schema-helper-and-json/MedcoMDSchema.json>`_
+- View the `helper code <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-json-schema-helper-and-json/HelperCode.java>`_ to generate the full JSON Schema for this MedcoMD guide in java. Note: this helper code is not the full helper code for this guide, it only contains the code to generate a JSON Schema for this section. 
```

Comment:
(I am guessing that github markdown broke your ordered list formatting and that the last three items are actually 3, 4, and 5)

1 - Thanks!
2a - Thanks!
2b - The comment was to use `createJSONSchema`, not `getJSONSchema` as `get` and `set` are usually used in the getter/setter pattern. Also, did you get a chance to talk to the Java drivers team about the builder pattern for the JSON Object that holds the schema?
2c - Sorry I meant *data key*. I was referencing this line:
```System.out.println(getJSONSchema("<paste_your_key_id_here>"));```

3 - Thanks!
5 - Ok, thanks. Is the code that loads the JSON Schema from the file be in the "Configure the Client" section? Or are we going to use `FieldLevelEncryptionJSONSchema.java` for that section?

Reply:
2b. I've switched it to createJSONSchema now. Also, I messaged Jeff from Java drivers team about builder pattern and waiting for his response.
2c. I see. I've add in an additional comment in the helper code (" // replace the "paste_your_key_id_here" with your data key id") in case "<paste_your_key_id_here>" was not clear enough
5. Going to defer to @nlarew on this one

Reply:
2c. That seems more verbose, but I'll leave it up to you.

Reply:
I'm going to keep it "<paste_your_key_id_here>". I'd take clarity > less verbosity in this use case